### PR TITLE
fix: Updates to Codey RAG Notebook

### LINKED ISSUE #409: code_retrieval_augmented_generation.ipynb: metaclass conflict
When running the unmodified `code_retrieval_augmented_generation.ipynb` notebook, the first code cell under Step 3 *Chunk & generate embeddings for each code strings & initialize the vector store* generates an error on line 16 related to the usage of metaclasses.

```
TypeError                                 Traceback (most recent call last)
[<ipython-input-11-7a0a3102dbd4>](https://localhost:8080/#) in <cell line: 16>()
     14 
     15 
---> 16 class CustomVertexAIEmbeddings(VertexAIEmbeddings, BaseModel):
     17     requests_per_minute: int
     18     num_instances_per_batch: int

TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

Removing the extra subclassing of `BaseModel` seems to resolve the issue.

### PR DESCRIPTION
**Pull request by holtskinner**
Fixes #409 🦕

- Other Formatting and Readability improvements

### CODE REVIEW

File: language/code/code_retrieval_augmented_generation.ipynb [hunk 1]
```diff
@@ -1,785 +1,773 @@
...
+        "# limitations under the License."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "XNPE46X8mJj4"
+      },
+      "source": [
+        "# Use Retrieval Augmented Generation (RAG) with Codey API's\n",
```
Comment (koverholt) [inline language/code/code_retrieval_augmented_generation.ipynb:32]: `API's` -> `APIs`

File: language/code/code_retrieval_augmented_generation.ipynb [hunk 2]
```diff
@@ -1,785 +1,773 @@
...
+        "\n",
+        "<table align=\"left\">\n",
```
Comment (koverholt) [inline language/code/code_retrieval_augmented_generation.ipynb:34]: This is rendering oddly in nbreview, not sure if it's just a nbreview rendering issue, or your table seems to be missing `<tr>` tags?

Screenshot 2024-02-22 at 11 46 22 AM

File: language/code/code_retrieval_augmented_generation.ipynb [hunk 3]
```diff
@@ -1,785 +1,773 @@
...
+        "\n",
+        "### Cost\n",
+        "\n",
+        "This tutorial uses billable components of Google Cloud:\n",
+        "\n",
+        "- Vertex AI PaLM APIs offered by Google Cloud\n",
+        "\n",
+        "Learn about [Vertex AI pricing](https://cloud.google.com/vertex-ai/pricing) and use the [Pricing Calculator](https://cloud.google.com/products/calculator/) to generate a cost estimate based on your projected usage.\n",
+        "\n",
+        "**Note:** We are using local vector store(FAISS) for this example however recommend managed highly scalable vector store for production usage such as [Vertex AI Matching Engine](https://cloud.google.com/vertex-ai/docs/vector-search/overview) or [AlloyDB for PostgreSQ](https://cloud.google.com/alloydb/docs/ai/work-with-embeddings) or [Cloud SQL for PostgreSQL](https://cloud.google.com/sql/docs/postgres/features)  using pgvector extension."
```
Comment (koverholt) [inline language/code/code_retrieval_augmented_generation.ipynb:101]: `PostgreSQ` -> `PostgreSQL`

File: language/code/code_retrieval_augmented_generation.ipynb [hunk 4]
```diff
@@ -1,785 +1,773 @@
...
+        "code_files_urls[0:10]"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "mFNVieLnR8Ie"
+      },
+      "source": [
+        "### 2. Extract code and from the Jupyter notebooks.\n",
```
Comment (koverholt) [inline language/code/code_retrieval_augmented_generation.ipynb:388]: Should be "Extract code from the Jupyter notebooks"

File: language/code/code_retrieval_augmented_generation.ipynb [hunk 5]
```diff
@@ -1,785 +1,773 @@
...
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "o537exyZk9DI"
+      },
+      "source": [
+        "Next we need to create Github personal token to be able to list all files in a repository.\n",
```
Comment (koverholt) [inline language/code/code_retrieval_augmented_generation.ipynb:267]: Should be "Next we need to create a GitHub personal token"

File: language/code/code_retrieval_augmented_generation.ipynb [hunk 6]
```diff
@@ -1,785 +1,773 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "T1AF3fhBSLOm"
+      },
+      "source": [
+        "### 3. Chunk & generate embeddings for each code strings & initialize the vector store\n",
+        "\n",
+        "We need to split code into usable chunks that LLM can use for code generation. Therefore its crucial to use the right chunking approach and chunk size."
```
Comment (koverholt) [inline language/code/code_retrieval_augmented_generation.ipynb:482]: Should be "that the LLM can use".

And `its` -> `it's`.

**Review by koverholt — APPROVED** [review]

File: language/code/code_retrieval_augmented_generation.ipynb (hunk 2 above)
Comment (holtskinner) [reply language/code/code_retrieval_augmented_generation.ipynb:34]: That's just an NBReview issue

**Pull request merged**